mongodb_ingestion/src/main/java/swim/vehicle/MongoDbPollingAgent.java

```diff
@@ -0,0 +1,37 @@
+package swim.vehicle;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCursor;
+import org.bson.Document;
+import swim.api.agent.AbstractAgent;
+import swim.json.Json;
+import swim.structure.Value;
+
+public class MongoDbPollingAgent extends AbstractAgent {
+
+  private FindIterable<Document> find() {
+    return Assets.mongoClient().getDatabase("myDatabase")
+        .getCollection("myCollection")
+        .find();
+  }
+
+  private void poll() {
+    try (MongoCursor<Document> cursor = find().cursor()) {
+      while (cursor.hasNext()) {
+        processDocument(cursor.next());
+      }
+    }
+  }
+
+  private void processDocument(final Document document) {
+    final Value body = Json.parse(document.toJson());
+    final String nodeUri = "/vehicle/" + body.get("id").longValue();
+    command(nodeUri, "addMessage", body);
+  }
+
+  @Override
+  public void didStart() {
+    asyncStage().task(this::poll).cue();
```

Comment:
`asyncStage().task(this::poll)` is shorthand for creating a new `AbstractTask` and only overriding `runTask`, i.e. using the default impls for all other methods including `taskWillBlock()` which by default returns `false`. I believe we would need `taskWillBlock()` to return `true` here. If that's the case, the long syntax is unfortunately required.